test/unit/routers/alpha-router/functions/compute-all-routes.test.ts

```diff
@@ -83,6 +101,17 @@ describe('compute all v2 routes', () => {
     expect(routes).toHaveLength(1);
   });
 
+  test('succeeds to compute all routes with 5 hops. ignoring arbitrage opportunities', async () => {
+    const pools = [DAI_USDT, USDC_DAI, USDC_USDT, USDC_WETH, WETH_USDT, WBTC_WETH];
+    const routes = computeAllV2Routes(USDC, WRAPPED_NATIVE_CURRENCY[1]!, pools, 5);
+
+    routes.forEach((route) => {
+      expect(route.pairs).not.toEqual([USDC_USDT, DAI_USDT, USDC_DAI, USDC_WETH]);
+      expect(route.pairs).not.toEqual([USDC_DAI, DAI_USDT, USDC_USDT, USDC_WETH]);
```

Comment:
can you add a test for mixed routes that it prevents something like

V2 (USDC -> DAI)  -> V3 (DAI -> USDC) -> V3 (USDC-> WETH) ?

I think i've seen that happen before